appIos/feed/feed/FeedScreen.swift

```diff
@@ -112,6 +120,10 @@ private struct FeedState: View {
                     }
                 }
             }
+            .onChange(of: searchQuery) { _ in
+                onSearchChanged(searchQuery)
+            }
+            .searchable(text: $searchQuery, prompt: "Search \"\(searchQuery.lowercased())\"")
```

Comment:
By doing this move of the `searchQuery` down, it is no longer persistent, I guess I can use a callback similar to `onSearchChanged` to return the value stored in the `state` of the view-model (or even make `onSearchChanged` have a side-effect of returning the view-model `state` value, but it does sound like multiple side-effects already...)

Reply:
I did this approach by adding `currentSearchQuery` to the `FeedState`

Reply:
This is exactly how it should be, nice one! We pass the current value and we have a change listener callback to avoid persisting state within the component.